Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -508,10 +510,80 @@ extension BrowserViewController: TopToolbarDelegate {
+        guard let url = shieldsViewController.tab.url else { return }
+        self?.showSubmitReportView(for: url)
+      }
+    }
+    
     let container = PopoverNavigationController(rootViewController: shields)
     let popover = PopoverController(contentController: container, contentSizeBehavior: .preferredContentSize)
     popover.present(from: topToolbar.locationView.shieldsButton, on: self)
   }
+  
+  private func showSubmitReportView(for url: URL) {
+    // Strip fragments and query params from url
+    var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+    components?.fragment = nil
+    components?.queryItems = nil
+    guard let cleanedURL = components?.url else { return }
+    
+    let viewController = UIHostingController(rootView: SubmitReportView(
+      url: cleanedURL,
+      submit: { [weak self] url, additionalDetails, contactInfo in
+        guard let self = self else { return }
+        let popover = PopoverController(
+          contentController: SubmitReportSuccessViewController(), 
+          contentSizeBehavior: .preferredContentSize
+        )
+        
+        let domain = Domain.getOrCreate(forUrl: url, persistent: !privateBrowsingManager.isPrivateBrowsing)
+        
+        let report = WebcompatReporter.Report(
+          cleanedURL: url,
+          additionalDetails: additionalDetails,
+          contactInfo: contactInfo,
+          areShieldsEnabled: !domain.areAllShieldsOff,
+          adBlockLevel: domain.blockAdsAndTrackingLevel,
+          fingerprintProtectionLevel: domain.finterprintProtectionLevel,
+          adBlockListTitles: FilterListStorage.shared.filterLists.compactMap({ return $0.isEnabled ? $0.entry.title : nil }),
+          isVPNEnabled: BraveVPN.isConnected
+        )
+        
+        popover.present(from: topToolbar.locationView.shieldsButton, on: self)
+        
+        Task { [weak self] in
+          await WebcompatReporter.send(report: report)
+          self?.delayDismiss(popover: popover)
+        }
+      }
+    ))
+    
+    viewController.modalPresentationStyle = .popover
+
+    if let popover = viewController.popoverPresentationController {
+      popover.sourceView = topToolbar.locationView.shieldsButton
+      popover.sourceRect = topToolbar.locationView.shieldsButton.bounds
+      
+      let sheet = popover.adaptiveSheetPresentationController
+      sheet.largestUndimmedDetentIdentifier = .medium
+      sheet.prefersEdgeAttachedInCompactHeight = true
+      sheet.widthFollowsPreferredContentSizeWhenEdgeAttached = true
+      
+      // Allows the user to reference the sheet
+      sheet.detents = [.medium(), .large()]
+    }
+    navigationController?.present(viewController, animated: true)
+  }
+  
+  private func delayDismiss(popover: PopoverController) {
+    Timer.scheduledTimer(withTimeInterval: 4, repeats: false) { _ in
+      guard !popover.isBeingDismissed else { return }
+      popover.dismissPopover()
```

Comment:
You're still retaining the popover until the timer ends, so it'll stay alive for 4 seconds regardless.

Reply:
I moved the logic to the view so I hope the sleep is no longer a problem. I'm trying to avoid timers because they don't seem to work well with Tasks